pkg/event/types.go

```diff
@@ -33,11 +33,12 @@ type Parser interface {
 // Context contains the metadata for an event
 // The fields are based on the CloudEvent spec: https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md
 type Context struct {
-	ID      ID        // required
-	Source  Source    // required
-	Type    Type      // required
-	Subject string    // optional
-	Time    time.Time // optional
+	ID      ID                // required
+	Source  Source            // required
+	Type    Type              // required
+	Subject string            // optional
+	Time    time.Time         // optional
+	Labels  map[string]string // optional
```

Comment:
Prior to the introduction of `Labels`, all of the `Context`'s `With()` funcs would return a pure copy. But since a `map` is _basically_ a pointer, the copies returned by `With()` functions will all share the same map - changing one will update all the others.

So we'll need to copy the map in all of these 😞